Framework/Algorithms/src/XDataConverter.cpp

```diff
@@ -103,7 +103,7 @@ void XDataConverter::setXData(API::MatrixWorkspace_sptr outputWS,
         }
       }
     }
-    outputWS->setX(index, m_cachedX);
+    outputWS->setSharedX(index, m_cachedX);
   } else {
     const MantidVec &xBoundaries = inputWS->readX(index);
     MantidVec &xPoints = outputWS->dataX(index);
```

Comment:
@PranavBahuguna Actually, you can do even better:

Change the signature of the conversion function to:

```cpp
Kernel::cow_ptr<HistogramX> XDataConverter::calculateXPoints(Kernel::cow_ptr<HistogramX> inputX) const;
```

You can then directly use the constructors of `Points` and `BinEdges` to to the work, instead of using the functions in `VectorHelper` (the latter will be removed soon), for example:

```cpp
Kernel::cow_ptr<HistogramX> ConvertToPointData::calculateXPoints(Kernel::cow_ptr<HistogramX> inputX) const {
  return Points(BinEdges(std::move(inputX))).cowData();
}
```

At the call site use something like

```cpp
outputWS->setSharedX(i, calculateXPoints(inputWS->sharedX()));
```

I think this should not change the performance results, since there will be no additional copy.

If you have trouble understanding this, don't hesitate to contact me on Slack, or get in touch with @LamarMoore, if you want to discuss this with someone fact to face.